src/wallet/spend.cpp

```diff
@@ -572,8 +585,8 @@ std::optional<SelectionResult> SelectCoins(const CWallet& wallet, const std::vec
     if (!res) return std::nullopt;
 
     // Add preset inputs to result
-    res->AddInput(preset_inputs);
-    if (res->m_algo == SelectionAlgorithm::MANUAL) {
+    res->Merge(preselected);
```

Comment:
I have the impression that this commit makes changes to how coin selection works with preselected inputs and in addition introduces that SelectionResult returns change. While both changes look worthwhile to me, it would be easier to follow if they were separate commits if not separate PRs. Am I missing some special reason why doing all of this at once is advantageous?

Reply:
I intended this to not change the behavior. The reason why we need `Merge` is to correctly track `SelectionResult::m_target`. Right now in case with preselected inputs `m_target` will be reduced by the amount of preselected inputs and this leads to incorrect result in `SelectionResult::GetChange`

I'll split the commits and try to explain better why this is needed

Reply:
Thanks